magicctapipe/utils/functions.py

```diff
@@ -1,294 +1,317 @@
-    event_coord: astropy.coordinates.sky_coordinate.SkyCoord
-        Coordinate of the shower arrival direction
-    tel_coord: astropy.coordinates.sky_coordinate.SkyCoord
-        Coordinate of the telescope pointing direction
+    pointing_ra: astropy.units.quantity.Quantity
+        Right ascension of the telescope pointing direction
+    pointing_dec: astropy.units.quantity.Quantity
+        Declination of the telescope pointing direction
+    on_coord_ra: astropy.units.quantity.Quantity
+        Right ascension of the center of the ON region
+    on_coord_dec: astropy.units.quantity.Quantity
+        Declination of the center of the ON region
     n_off_regions: int
         Number of OFF regions to be extracted
 
     Returns
     -------
-    theta_on: astropy.units.quantity.Quantity
-        Angular distance from the ON region
-    theta_off: dict
-        Angular distances from the OFF regions
     off_coords: dict
-        Coordinates of the OFF regions
+        Coordinates of the centers of the OFF regions
     """
 
-    # Compute the distance from the ON region:
-    theta_on = on_coord.separation(event_coord)
+    ra_diff = pointing_ra - on_coord_ra
 
-    # Compute the wobble offset and rotation:
-    offsets = np.arccos(
-        np.cos(on_coord.dec)
-        * np.cos(tel_coord.dec)
-        * np.cos(tel_coord.ra - on_coord.ra)
-        + np.sin(on_coord.dec) * np.sin(tel_coord.dec)
+    # Calculate the wobble offset
+    wobble_offset = np.arccos(
+        np.cos(on_coord_dec) * np.cos(pointing_dec) * np.cos(ra_diff)
+        + np.sin(on_coord_dec) * np.sin(pointing_dec)
     )
 
-    numerator = np.sin(tel_coord.dec) * np.cos(on_coord.dec) - np.sin(
-        on_coord.dec
-    ) * np.cos(tel_coord.dec) * np.cos(tel_coord.ra - on_coord.ra)
-
-    denominator = np.cos(tel_coord.dec) * np.sin(tel_coord.ra - on_coord.ra)
-
-    rotations = np.arctan2(numerator, denominator)
-    rotations[rotations < 0] += u.Quantity(360, u.deg)
+    # Calculate the wobble rotation angle
+    numerator_1 = np.sin(pointing_dec) * np.cos(on_coord_dec)
+    numerator_2 = np.cos(pointing_dec) * np.sin(on_coord_dec) * np.cos(ra_diff)
+    denominator = np.cos(pointing_dec) * np.sin(ra_diff)
 
-    # Define the wobble and OFF coordinates:
-    mean_offset = offsets.to(u.deg).mean()
-    mean_rot = rotations.to(u.deg).mean()
+    wobble_rotation = np.arctan2(numerator_1 - numerator_2, denominator)
+    wobble_rotation = Angle(wobble_rotation).wrap_at(360 * u.deg)
 
-    skyoffset_frame = SkyOffsetFrame(origin=on_coord, rotation=-mean_rot)
+    # Calculate the OFF coordinates
+    wobble_coord = SkyCoord(pointing_ra, pointing_dec, frame="icrs")
 
-    wobble_coord = SkyCoord(mean_offset, u.Quantity(0, u.deg), frame=skyoffset_frame)
-    wobble_coord = wobble_coord.transform_to("icrs")
+    rotations_off = np.arange(0, 359, 360 / (n_off_regions + 1)) * u.deg
+    rotations_off = rotations_off[rotations_off.to_value(u.deg) != 180]
```

Comment:
why is 180 excluded? I would rather expect to exclude 0

Reply:
This is because, in this function the 180 deg is the same position as the ON source.

Reply:
For more information this is how I understand. If we add 180 deg to the wobble rotation angle, the OFF coordinate is created at the same position as the source coordinate.
![coordinates](https://user-images.githubusercontent.com/39613098/192425418-2214fa8e-5c24-4f97-a42b-7a364d81f0c6.png)

Reply:
I added a small comment on this code.

Reply:
Ok, now I see, I thought that you were defining wobble_rotation differently